### ISSUE TITLE
Small issue with urls from \Phalcon\Mvc\Url

### ISSUE THREAD

**Issue by dompie** [phalcon/cphalcon#3222]
In the router I use:

```php
$router->addGet('/activation/{activationToken:([a-zA-Z0-9_]{19,45})}', array(
	'controller' => 'index',
	'action' => 'activation',
))->setName('activationUrl');
```

When generating an URL for this action with the following code:

```php
$url = Di::getDefault()->get('url'); // This is \Phalcon\Mvc\Url
$activationUrl = $url->get(array(
	'for' => 'activationUrl',
	'activationToken' => $activationToken
));
```
The generated url looks like this for base uri `'http://' . $_SERVER['SERVER_NAME']` (which is "www.local.dev"):

`http://www.local.devactivation/u9o6xzwql3pirc42k9m04fv`

And I can not add a trailing slash after $_SERVER['SERVER_NAME'], because this leads to multiple slashes in some cases.
Can this be fixed by reconfiguration?

I use phalcon 1.3.4

**Comment by dompie** [comment]
Could workaround this issue by changing:
```php
$router->addGet('/activation/{activationToken:([a-zA-Z0-9_]{19,45})}', array(
```
to
```php
$router->addGet('//activation/{activationToken:([a-zA-Z0-9_]{19,45})}', arr...
```

Other suggestions / solutions welcome.

**Comment by 7thcubic** [comment]
what did you set up for ur base url?

$di->set('url', function () use (&$config) {
    $url = new UrlResolver();
    $url->setBaseUri($config->application->baseUri); // << this one
    return $url;
}, true);

**Comment by dompie** [comment]
I use $_SERVER['SERVER_NAME'] for baseUri

**Comment by 7thcubic** [comment]
If you configured your base url to be

http://.$_SERVER["SERVER_NAME"]."/"

when ever you use the functionality from url. do not include the prevailing slash.
eg.

$url->get("index/login");

if its http://.$_SERVER["SERVER_NAME"]

do it as 

$url->get("/index/login");

so you really have to set the standard of coding.

**Comment by dompie** [comment]
That's what I'm saying: I already use 
```php
define('BASE_URI', 'http://' . $_SERVER['SERVER_NAME']);
...
$url->setBaseUri(BASE_URI);
```
and I define all my routes like this *always*

```php
$router->addGet('/activation/{activationToken:([a-zA-Z0-9_]{19,45})}', array(
	'controller' => 'index',
	'action' => 'activation',
))->setName('activationUrl');

$router->addGet('/activation/resend/{identifier:([0-9]{6,18})}', array(
	'controller' => 'index',
	'action' => 'resendActivation'
))->setName('resendActivation');

```
And plahcon generates "http://www.local.devactivation/u9o6xzwql3pirc42k9m04fv" instead of "http://www.local.dev/activation/u9o6xzwql3pirc42k9m04fv"

**Comment by sergeyklay** [comment]
Base uri __always__ must end with `/`

**Comment by 7thcubic** [comment]
dompie i found out the issue.

Router when u add the path u have to create it with a "/" infront

Meaning your router add get is correct.

As for base uri, please set it with a trailing slash.

for a simple $router->get("link"); dont place the slash infront.

by default the pattern matching removes the first slash infront of the pattern.

**Comment by sergeyklay** [comment]
```php
        $config = $this->di['config'];
        $baseUrl = $config->application->baseUri;
        $routes = $config->routes;

        $this->di['url'] = function () use ($baseUrl) {
            $url = new Phalcon\Mvc\Url;
            $url->setBaseUri($baseUrl);
            return $url;
        };

        $this->di['router'] = function () use ($routes) {
            $router = new \Phalcon\Mvc\Router;

            $router->removeExtraSlashes(true); // DO NOT FORGET ABOUT IT

            // ...
            // ...


            return $router;
        };
```

**Comment by dompie** [comment]
Yes, this works good now. I have baseUri with "'mydomain.com' . '/'" (slash at the end), set "removeExtraSlashes" to true and added all routes with a starting slash. Generated urls from router are correct.

I call "get()" this way:
```php
$this->url->get(array(
	'for' => 'activationUrl',
	'activationToken' => $activationToken
));
```

But there is still one thing left. When I do a redirect to '/' ``` $this->response->redirect('/');``` then the resulting URL will look like 'http://mydomain.com//'.
It's not a big thing, but an indication that something is still not right.

**Comment by olivier-monaco** [comment]
@dompie Can you try with latest version please?

**Comment by Green-Cat** [comment]
@andresgutierrez Fixed in #3315
